src/main/groovy/com/ullink/AssemblyInfoVersionPatcher.groovy

```diff
@@ -13,7 +13,7 @@ class AssemblyInfoVersionPatcher extends ConventionTask {
     List<String> projects = []
 
     AssemblyInfoVersionPatcher() {
-        conventionMapping.map "projects", { [ project.tasks.msbuild.mainProject?.projectName ] }
+        conventionMapping.map "projects", { project.tasks.msbuild.projects.collect { it.name } }
```

Comment:
But I think it would be better than patch only the main project as a solution could have more than one projects, if the test project has no version specified then it would be patched in the end?